src/mono/wasm/runtime/library_mono.js

```diff
@@ -779,6 +779,49 @@ var MonoSupportLib = {
 			// and nested class names like Foo/Bar to Foo.Bar
 			return className.replace(/\//g, '.').replace(/`\d+/g, '');
 		},
+
+		mono_wasm_load_data: function (data, prefix, magic) {
```

Comment:
I think we should hardcode the magic number here (but it looks reversed?).  If it isn't hardcoded and isn't defined it should skip the check entirely though